content/en/tutorials/securely_accessing_network_locally.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Securely Accessing Opta Networks Locally"
+linkTitle: "Securely Accessing Opta Networks Locally"
+date: 2022-01-03
+draft: false
+description: >
+  Instructions to integrate with the CI/CD platform of your choice
+---
+
+# Overview
+When working with secure private networks, and specifically with the databases in them, one may wish to somehow manually 
+connect to them for debugging purpose, but there is This action, naturally, has security concerns and there's many 
+solutions out there. For our users, we typically recommend installing VPN servers within your opta-managed VPCs and 
+handling network access to your environments from there. For just getting started, however, there's a rather 
+straight-forward TCP proxy solution. This article will discuss both solutions and the security concerns.
+
+# TCP-Proxy Service
+A quick way to gain connectivity is to create a TCP-Proxy service with opta and using `kubectl port-forward` to gain
+access to the proxy locally. Basically, there is this [docker image called tecnativa/tcp-proxy](https://hub.docker.com/r/tecnativa/tcp-proxy)
+and it pretty much does what its name says: based on environment variables specified by the user, it forwards all tcp
+traffic sent to it on a user-specified port to another user-specified host + port. The way that you can leverage this
+with opta is by creating a new k8s-service with Opta using that image, and forwarding TCP connections to your database.
+
+So suppose you have a postgres database (so port 5432, for redis it would be 6379, for mysql 3306 etc...) and its host 
+is 10.40.153.14 (you can find out by checking the envars in a linked service). If you wanted to connect to it locally,
+you could do so by first creating a following service in your opta environment:
+```yaml
+environments:
+  - name: my-env-yaml
+    path: "../my-env-yaml.yaml"
+name: mytcpproxy
+modules:
+  - name: app
+    type: k8s-service
+    image: tecnativa/tcp-proxy
+    min_containers: 1
+    max_containers: 1
+    port:
+      tcp: 5432
+    env_vars:
+      TALK: "10.40.153.14:5432"
+      LISTEN: ":5432"
+```
+
+After that service is created you can do a port forward to your localhost with the kubectl port-forward command:
+
+```kubectl port-forward -n mytcpproxy svc/app 5432:80```
```

Comment:
where is 80 coming from? It looks like the `LISTEN` env is set to `5432`?

Reply:
opta service always uses port 80, but this should be invisible to the user

Reply:
oh got it.